### PR TITLE
chore: upgrade to v10

### PR DESCRIPTION
**Pull request by Cyberxon**
## Summary

Upgrade master to v10 with the changes in [beta-v10](https://github.com/contentful/contentful.js/tree/beta-v10)

## Steps

1. Merge Beta-v10
2. Delete remaining .js files
3. Cleanup one file after prettier check

### CODE REVIEW

File: DOCS.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<p align="center">
+  <a href="https://www.contentful.com/developers/docs/references/content-delivery-api/">
+    <img alt="Contentful Logo" title="Contentful" src="media://contentful-icon.png" width="150">
```
Comment (marcolink) [inline DOCS.md:3]: For me the logo is not loading here: https://github.com/contentful/contentful.js/blob/chore/upgrade-to-v10/DOCS.md is that expected?

Comment (Cyberxon) [reply DOCS.md:3]: it's for the generated typedoc, it's not meant to be rendered on github

File: package.json [hunk 2]
```diff
@@ -4,8 +4,32 @@
...
+    "compilerOptions": {
+      "rootDir": "./lib",
+      "typeRoots": [
+        "./lib"
+      ]
+    }
+  },
+  "files": [
+    "dist",
+    "lib",
```
Comment (marcolink) [inline package.json:23]: I'm not sure - should we include source files? 🤔

**Review by marcolink — APPROVED** [review]
Big moment 🎉

**Pull request merged**